src/population.hpp

```diff
@@ -121,17 +122,74 @@ std::vector<T> _readSelection(const HighFive::DataSet& dset, const Selection& se
     return result;
 }
 
+namespace collective {
+template <typename T>
+std::vector<T> _readSelection(const HighFive::DataSet& dset,
+                              const Selection& selection,
+                              const DataTransferOpts& data_transfer_opts) {
```

Comment:
This is one example of the changes required to `libsonata` to make it save for collective IO. Similar changes would be applied to two overloads of `bbp::sonata::_readSelection`; and anything in `edge_index.cpp`.